### PR TITLE
Make sure building with relative paths work correctly.

### PR DESCRIPTION
**Pull request by rhatdan**
Fixes: https://github.com/containers/podman/issues/12763

Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

File: pkg/bindings/images/build.go [hunk 1]
```diff
@@ -352,11 +352,13 @@ func Build(ctx context.Context, containerFiles []string, options entities.BuildO
 			}
 			c = tmpFile.Name()
 		}
+		c = filepath.Clean(c)
 		cfDir := filepath.Dir(c)
 		if absDir, err := filepath.EvalSymlinks(cfDir); err == nil {
 			name := filepath.ToSlash(strings.TrimPrefix(c, cfDir+string(filepath.Separator)))
 			c = filepath.Join(absDir, name)
 		}
+
```
Comment (flouthoc) [inline pkg/bindings/images/build.go:361]: nit: extra line

**Comment by flouthoc** [comment]
@Luap99 Some of the unrelated netavark tests are failing, could it be that we have regressed netavark upstream ?

**Comment by vrothberg** [comment]
> @Luap99 Some of the unrelated netavark tests are failing, could it be that we have regressed netavark upstream ?

Are we merging netavark PRs without exercising them here?

**Comment by flouthoc** [comment]
@vrothberg Afaik I think yes but i am not sure if changes are immediately reflected on podman's CI. @cevich  @Luap99 Could tell this better.

**Comment by vrothberg** [comment]
> @vrothberg Afaik I think yes but i am not sure if changes are immediately reflected on podman's CI. @cevich @Luap99 Could tell this better.

If that's the case, Podman CI should probably pick specific commits from netavark (instead of branch which I presume is what we're doing). Just as we do with all other dependencies and building blocks.

**Comment by Luap99** [comment]
Yes we use the main branch currently, The plan is to switch to the packaged netavark once we have a release.
I do not understand why we get timeouts now, the only netavark PR that was merged is this one: https://github.com/containers/netavark/pull/218/files and I do not think this causes a hang.
We also made some aardvark changes but I do not see why this would cause a hang since aardvark runs in the background.

**Comment by Luap99** [comment]
My netavark PR did cause the issue, I accidentally swapped the logic there.
https://github.com/containers/netavark/pull/220 will fix it, once merged you need to wait ~15min and then you can restart the tests. A force push is not required.

**Comment by jwhonce** [comment]
LGTM

**Review by flouthoc — APPROVED** [review]

**Comment by cevich** [comment]
> Are we merging netavark PRs without exercising them here?

> Afaik I think yes but i am not sure if changes are immediately reflected on podman's CI. @cevich @Luap99 Could tell this better.

FYI (since this is already merged), yes this is the case.  It's likely a signal of a testing gap in netavark CI.  We could (in theory) run some select podman e2e tests under netavark CI, but it's not a small amount of work to get that going.

**Comment by vrothberg** [comment]
> FYI (since this is already merged), yes this is the case. It's likely a signal of a testing gap in netavark CI. We could (in theory) run some select podman e2e tests under netavark CI, but it's not a small amount of work to get that going.

We're doing something similar in Skopeo and c/image.  c/image always uses a branch of Skopeo (because those are the actual tests) but Skopeo uses a _commit_ of c/image to make vendoring explicit and from another repo breaking CI.

**Pull request merged**